### ISSUE TITLE
System.IO.FileLoadException after upgrade to 2.3.0-rc1 with netcoreapp1.0 and 1.1

### ISSUE THREAD

**Issue by GeorgDangl** [xunit/xunit#1471]
A repro project can be found at https://github.com/GeorgDangl/Dangl.Common/commit/611839fbf4df8ef524bcd410de5cc17f3829bb53 in the test project.

The [gist of the issue](https://gist.github.com/GeorgDangl/d426ab7890bfd304a05c5e8458adba0d):

`dotnet xunit -f netcoreapp1.0` produces the following output in individual tests

    System.IO.FileLoadException : Could not load file or assembly
    'System.IO.Compression, Version=4.1.1.0,
    Culture=neutral, PublicKeyToken=b77a5c561934e089'.
    Could not find or load a specific file. (Exception from HRESULT: 0x80131621)

`dotnet xunit -f netcoreapp1.1` does not run any tests at all but outputs:

    Running .NET Core tests for framework netcoreapp1.1...
    xUnit.net Console Runner (64-bit .NET Core 4.0.0.0)
    System.IO.FileLoadException: Could not load file or assembly
    'System.Runtime.InteropServices, Version=4.1.1.0,
    Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
    Could not find or load a specific file. (Exception from HRESULT: 0x80131621)

Downgrading to `2.3.0-beta5-build3769` fixes the issues. Also, all tests run fine with `dotnet test`. Full .Net framework and `netcoreapp2.0` run fine.

**Comment by bradwilson** [comment]
## netcoreapp1.0

The problem with `netcoreapp1.0` is this line:

https://github.com/GeorgDangl/Dangl.Common/blob/491f9b1633fe702adc4a66c3557e6009a6e162b3/src/Dangl.Common/Dangl.Common.csproj#L26

```xml
<NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.3' ">1.6.1</NetStandardImplicitPackageVersion>
```

.NET Core 1.0 requires 1.6.0 of this package, not 1.6.1. Why did you add this line?

## netcoreapp1.1

The problem with `netcoreapp1.1` is that it's trying to run the tests with .NET Core 1.0. I'm able to make the tests run correctly with `dotnet xunit -framework netcoreapp1.1 -fxversion 1.1.1`:

```
PS> dotnet xunit -framework netcoreapp1.1 -fxversion 1.1.1
Detecting target frameworks in Dangl.Common.Tests.csproj...
Building for framework netcoreapp1.1...
  Dangl.Common -> C:\Dev\repros\xunit-1471\src\Dangl.Common\bin\Debug\netstandard1.3\Dangl.Common.dll
  Dangl.Common.Tests -> C:\Dev\repros\xunit-1471\test\Dangl.Common.Tests\bin\Debug\netcoreapp1.1\Dangl.Common.Tests.dll
Running .NET Core tests for framework netcoreapp1.1...
xUnit.net Console Runner (64-bit .NET Core 4.6.25009.03)
  Discovering: Dangl.Common.Tests
  Discovered:  Dangl.Common.Tests
  Starting:    Dangl.Common.Tests
  Finished:    Dangl.Common.Tests
=== TEST EXECUTION SUMMARY ===
   Dangl.Common.Tests  Total: 84, Errors: 0, Failed: 0, Skipped: 0, Time: 0.303s
```

Any thoughts on why this is, @onovotny @natemcmaster? Should we (specifically `dotnet-xunit`) be proactively trying to figure out what framework version they're linked against? If so, is this something I can discover [here](https://github.com/xunit/xunit/blob/c7140cbc14a0d190931add88c2c0be9f3f2de440/src/dotnet-xunit/Program.cs#L267-L315)?

**Comment by bradwilson** [comment]
Using `dotnet msbuild /pp` I was able to determine that `RuntimeFrameworkVersion` will always get set, so I should be able to use that as a default value for `-fxversion`.

**Comment by bradwilson** [comment]
After upgrading to `2.3.0-rc1-build3807` (which contains the fix for #1476), `netcoreapp1.1` is working, but `netcoreapp1.0` is still failing because of the aforementioned line in your csproj file. When I change `NetStandardImplicitPackageVersion` to `1.6.0`, then everything works as expect:

![dotnet xunit](https://user-images.githubusercontent.com/16944/31058561-83d79bc4-a6aa-11e7-91dd-590c5fc2eafc.png)

I don't have a lot of insight as to why `dotnet test` works when you reference 1.6.1, other than a guess that it's actually using .NET Core 1.1 when running the tests (as when I use `-fxversion 1.1.1` to run your `netcoreapp1.0` tests, it works fine, because .NET Core 1.1 does support 1.6.1).

I'm going to consider this issue partially fixed (`netcoreapp1.1`) and partially "working as designed" (`netcoreapp1.0`).

**Comment by GeorgDangl** [comment]
I've just upgrade to xunit `2.3.0-rc1-build3808` and can confirm that it's working with `netcoreapp1.1`.

As for why this line is present:

    <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.3' ">1.6.1</NetStandardImplicitPackageVersion>

That's been added automatically when migrating from `xproj` to `csproj` with Visual Studio 2017. I've set it to `1.6.0`, just as you said, and everything is working.

Thank you very much!
